### DISCUSSION TITLE
Why does wezterm start a login shell by default ?

### DISCUSSION THREAD

**Discussion by dinkonin** [wezterm/wezterm#4544] [General]
I'm using zs, and use my zprofile te set some variables and execute some time consuming commands on every login. Wezterm always loads the .zprofile file which IMHO is not currect, no other terminal emulator (konsole, kitty, xterm, alacritty) does this.

Is there a reason for this or should I just do `config.default_prog = { '/usr/bin/zsh'}` in my config

**Comment by wez** [comment]
The reason is consistency of behavior between the mux server (which can be run independent from a separate login session) and the gui wezterm.  Regarding other terminal emulators: note that iterm and the macos terminal spawn login shells by default as well.

If you don't like it, you can set your default prog as you've shown above.

**Reply by dinkonin** [reply]
Thank you! I just wanted to be sure that I don't subtly break something with my aforementioned config.

**Reply by uwidev** [reply]
It should be noted that because it grabs the login shell, it will run all login-shell-related files. I had my graphical interface autoload on `.zprofile` and I spawned duplicates of my desktop on every shell I spawned. Not a major issue, but if someone declares environmental variables, it'll clog up their $PATH, among other things. Maybe something to consider adjusting?

**Reply by af3556** [reply]
To avoid hard-coding the shell in wezterm, the following seems to work well:
```
config.default_prog = { os.getenv("SHELL") },
```

**Comment by makarov-igor** [comment]
Sorry to ask beginner's question(s):

I run wezterm with config.default_prog = { '/usr/bin/zsh'}
but 
     $ echo $SHELL
still returns /bin/bash even if zsh is running.
and
     $ echo $0
is working as expected, returning '-zsh'

Won't some scripts with shell-specific parts fail on that ?
Is there a way to reassign $SHELL in wezterm config to 'zsh' ?

A peculiar observation: 
if I run wezterm with config.default_prog = { '/usr/bin/tmux'},
and have tmux pick zsh as the shell to run, then both $SHELL and $0 work as expected.

**Reply by wez** [reply]
`$SHELL` is defined as: `The user's shell, as specified by the password database.`

Unless you explicitly set `SHELL` via https://wezterm.org/config/lua/config/set_environment_variables.html or through the `env` key in the various spawn command helpers, wezterm will populate `SHELL` with the value from the password database in order to conform to the classic definition of `SHELL`.
